linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/Emitter.scala

```diff
@@ -482,17 +501,36 @@ final class Emitter(config: Emitter.Config, prePrinter: Emitter.PrePrinter) {
       }
     }
 
-    val isJSClass = kind.isJSClass
-
     // Class definition
-    if (linkedClass.hasInstances && kind.isAnyNonNativeClass) {
-      val isJSClassVersion = Version.fromByte(if (isJSClass) 1 else 0)
+    if (uncachedDecisions.hasInstances && kind.isAnyNonNativeClass) {
+      /* Decision-making in this scope is governed by the `fullClassChangeTracker`
+       * below instead of the uncachedDecisions. It is actually called at the
+       * end of the block, because it needs to track *produced* js.Trees in
+       * addition to *inputs* in the LinkedClass.
```

Comment:
Maybe we can clarify this a bit more? IIUC there are 2 reasons we need do this:

- Shape of the class (ES class or not)
- Method set.

Note the second one points to a problem: How do we make sure we trigger "changed" if the static method set changes?